services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceHelixAdmin.java

```diff
@@ -1174,6 +1182,70 @@ private Integer fetchSystemStoreSchemaId(String clusterName, String storeName, S
     return response.getId();
   }
 
+  static boolean isPushJobFailedUserError(PushJobDetailsStatus status, PushJobCheckpoints pushJobCheckpoint) {
+    if (PushJobDetailsStatus.isFailed(status)) {
+      if (pushJobCheckpoint != null) {
+        switch (pushJobCheckpoint) {
+          case QUOTA_EXCEEDED:
+          case WRITE_ACL_FAILED:
+          case DUP_KEY_WITH_DIFF_VALUE:
+          case INPUT_DATA_SCHEMA_VALIDATION_FAILED:
+          case EXTENDED_INPUT_DATA_SCHEMA_VALIDATION_FAILED:
+          case RECORD_TOO_LARGE_FAILED:
+          case CONCURRENT_BATCH_PUSH:
+          case DATASET_CHANGED:
+          case INVALID_INPUT_FILE:
+          case DVC_INGESTION_ERROR_DISK_FULL:
+          case DVC_INGESTION_ERROR_MEMORY_LIMIT_REACHED:
+            return true;
+
+          default:
+            return false;
+        }
+      }
+    }
+    return false;
+  }
+
+  static void emitPushJobDetailsMetrics(Map<String, PushJobStatusStats> pushJobStatusStatsMap, PushJobDetails value) {
+    List<PushJobDetailsStatusTuple> overallStatuses = value.getOverallStatus();
+    if (overallStatuses.isEmpty()) {
+      return;
+    }
+    boolean isIncrementalPush = false;
+    String cluster = value.getClusterName().toString();
+    PushJobStatusStats pushJobStatusStats = pushJobStatusStatsMap.get(cluster);
+    Utf8 incPushKey = new Utf8("incremental.push");
+    if (value.getPushJobConfigs().containsKey(incPushKey)) {
```

Comment:
There is a gap in VPJ where the `PushJobConfigs` get added to the `PushJobDetails` fairly late. I think we're going to need to fix this at some time. It need not be done in this PR

Reply:
Good call out. Currently haven't made any changes in the latest commit wrt this. Will take a look at this.

Reply:
Fixed this as part of this [commit](https://github.com/linkedin/venice/pull/1185/commits/e06fa3b137c4ba01d02ccc54440276880db6c240)